Automattic/gridicons
# Built: refactor tasks to file, demo template

![screen shot 2017-07-21 at 16 29 35](https://user-images.githubusercontent.com/4389/28470510-cf145d5e-6e31-11e7-8d95-69583ce734f7.png)

This PR:

* moves all the custom tasks to individual files so they are more easily reused and updated
* moves the template for the demo outside the Grunt file in its own template file

Why?

1. Easier to spot what changed in the build process
2. More easily reused across different repositories (Gridicons, Social Icons, Dashicons)
3. Cleaner separation of dependencies
4. Folder and files input declared in config instead of being hardcoded
5. Separate demo template
6. Separate react template

Note: there's also one icon added as PDF as it was added before we had the automated PDF build introduced in #238. 

### How to test

1. Switch to this branch
2. `grunt`
3. Check no changes to any built file anywhere, with the exception of the tabulation of the demo.
4. Check the compiled React component works.
